### ISSUE TITLE
OHLC candlestick width when using DateTime axis

### ISSUE THREAD

**Issue by ticool** [ScottPlot/ScottPlot#812]
Hi there nice work ! I like this package :)

I got 3 small thing that im not shure if im doing wright i lookup at the demo but it seems outdated for some method.
Im using .Net 5 project.

- First when im using Plot OHLC every thing works fine but when i change the plot mode to PlotCandleStick there is only verticals lines apprearing in the graph ?  Code looks like this :

```cs
formsPlot1.plt.PlotCandlestick(OHLCList.ToArray());
formsPlot1.plt.Ticks(dateTimeX: true, numericFormatStringY: "F7", displayTickLabelsX:false);
```
![image](https://user-images.githubusercontent.com/393970/108606121-424bae00-7386-11eb-9f8c-7fe68c29550b.png)

- I looked up about the Bollinger Bands and i got a vertical line in middle of the graph when i tried to use mutliple PlotScatter with OHLC. When im only using PlotScatter i can add 3 of them and its works fine.

 ```cs
double[] xs2 = xs.Skip(20).ToArray();
 (var sma, var bolL, var bolU) = ScottPlot.Statistics.Finance.Bollinger(OHLCList.ToArray(), 20);

 // plot technical indicators as scatter plots above the financial chart
formsPlot1.plt.PlotScatter(xs2, sma, color: Color.Blue, lineStyle: LineStyle.Solid);
 formsPlot1.plt.PlotScatter(xs2, bolL, color: Color.Red, lineStyle: LineStyle.Solid);
formsPlot1.plt.PlotScatter(xs2, bolU, color: Color.Green, lineStyle: LineStyle.Solid);
```

- My las point is about using PlottableText when i set the  .frameColor = Color.White or any other color the back grame color stay transparent ?

 ```cs
 var plottables = formsPlot1.plt.GetPlottables();
var scatterPlot = (PlottableOHLC)plottables[0];
var highlightText = (PlottableText)plottables[1];
highlightText.FontSize = 12;
highlightText.frameColor = Color.White;
```

looked at :
 https://swharden.com/scottplot/cookbooks/4.0.36/#plottypes-finance-bollinger-bands
And the demo bundle source

Thx if you can help me !!!

**Comment by swharden** [comment]
Hi @ticool, I'm happy you're finding ScottPlot helpful! I'll break my responses into 3 parts to make it easier to discuss each topic.

## Use ScottPlot 4.1-beta

First, I recommend upgrading to the latest version of ScottPlot. It's pre-release, which means you have to click the pre-release checkbox when searching NuGet for the package. Its cookbook is better, and some of these issues might be solved by using this newer version.

## Candlestick Width

This seems to be a common point of confusion, so I'll add this to the FAQ section. Briefly, OHLC objects have a `timeSpan` field you can assign. The units are in days:

```cs
foreach(var ohlc in ohlcs)
    ohlc.timeSpan = 1.0 / (24 * 60); // one minute (value is the width of the candle in days)
```

This is a common question so I'll add it to the FAQ https://github.com/ScottPlot/Website/issues/6 and I'll make a cookbook example too.

## Bollinger Bands are not Lining-Up with the Plot

I can't tell exactly what the problem is since you didn't share all your code. My guess is that your candles are plotted using `DateTime.ToOATime()` units but the Xs for your Bollinger Bands are not (or vise-versa?). Seeing the full code would be more helpful in figuring out what's up here. This may also be fixed in version 4.1

## Text Fill and Frame color
> My last point is about using PlottableText when i set the .frameColor = Color.White or any other color the back grame color stay transparent ?

Does this code example help?

```cs
var myText = plt.AddText("Test", 1, 2);
myText.Color = Color.Red;
myText.FontBold = true;
myText.BackgroundFill = true;
myText.BackgroundColor = Color.LightBlue;
```

**Comment by ticool** [comment]
Yea tons of new changes in 4.1.8 then 4.1.9 ! hehe

Yay nice candle stick :) for the Width you got it the problem came from Timespan im using this now :

```cs
private OHLC TransformCandle(Candle candle)
{
    DateTime date = candle.Time.UtcDateTime.ToLocalTime();
    TimeSpan timeSpan = TimeSpan.FromSeconds(GetGranularity);
    return new OHLC((double)candle.Open, (double)candle.High, (double)candle.Low, (double)candle.Close, date, timeSpan);
}
```

Bollinger Bands works like a charm with :
 
```cs
  double[] xs = OHLCList.Select(x => x.DateTime.ToOADate()).ToArray();
```

For the last point there is no more  formsPlot1_MouseMoved event and there seems to be alot of changes to do !

I took the code from : https://github.com/ScottPlot/ScottPlot/issues/181


Thx for quick help !
